nixos/modules/services/networking/ssh/sshd.nix

```diff
@@ -432,6 +432,7 @@ in
               { ExecStart =
                   (optionalString cfg.startWhenNeeded "-") +
                   "${cfgc.package}/bin/sshd " + (optionalString cfg.startWhenNeeded "-i ") +
+                  "-D " +  # don't detach into a daemon process
```

Comment:
I guess this is fine? https://github.com/archlinux/svntogit-packages/blob/af122946eba43779c5570e35c70c328fab69f5ee/trunk/sshd.service#L8

Reply:
Yes, that's the ultimate goal, to start sshd with `-D`, just like your ArchLinux example shows.

Reply:
I thought non-forking services required a different systemd service type option, and I see no change here? Is sshd written so that it signals to systemd when it's ready? Or didn't that work properly when it was forking either?

Reply:
Good question.

According to the manual[1], the Type option should be 'forking' for traditional unix daemons that fork and exit. However, the Type was already set to 'simple', that was confirmed as a small experiment shows that the `systemctl start sshd` hangs when type==forking. (even without the -D flag).

With type=simple, it works both with the -D and without.

However, specifying the -D (don't fork) flag makes it more compliant with the idea of Type=simple.

The -D also helps with alternative init systems, like my work with s6 [2].

1: https://www.freedesktop.org/software/systemd/man/systemd.service.html#Type=
2: https://sr.ht/~guido/nixos-init-freedom/

Reply:
Yes, -D should be correct with type simple.